### PR TITLE
Treat misconfigured rules as not passing

### PR DESCRIPTION
**Pull request by kgaillot**
This breaks behavioral backward compatibility and so will not be backported to the 2.1 branch

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , here's a compatibility break PR for review when you get a chance

File: lib/common/rules.c [hunk 1]
```diff
@@ -185,9 +183,10 @@ check_range(const xmlNode *date_spec, const char *id, const char *attr,
  * \param[in] now        Time to check
  *
  * \return Standard Pacemaker return code (specifically, EINVAL for NULL
- *         arguments, pcmk_rc_ok if time matches specification, or
- *         pcmk_rc_before_range, pcmk_rc_after_range, or pcmk_rc_op_unsatisfied
- *         as appropriate to how time relates to specification)
+ *         arguments, pcmk_rc_unpack_error if the specification XML is invalid,
+ *         pcmk_rc_ok if time matches specification, or pcmk_rc_before_range,
+ *         pcmk_rc_after_range, or pcmk_rc_op_unsatisfied as appropriate to how
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/rules.c:188]: thx

File: lib/common/rules.c [hunk 2]
```diff
@@ -320,7 +320,7 @@ pcmk__unpack_duration(const xmlNode *duration, const crm_time_t *start,
     ADD_COMPONENT(pcmk__time_minutes);
     ADD_COMPONENT(pcmk__time_seconds);
 
-    return rc;
+    return pcmk_rc_ok;
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/rules.c:323]: might as well

File: lib/common/rules.c [hunk 3]
```diff
@@ -1082,29 +1080,35 @@ pcmk__evaluate_attr_expression(const xmlNode *expression,
...
     // Get and validate reference value
     value = crm_element_value(expression, PCMK_XA_VALUE);
     switch (comparison) {
         case pcmk__comparison_defined:
         case pcmk__comparison_undefined:
-            if (value != NULL) {
-                pcmk__config_warn("Ignoring " PCMK_XA_VALUE " in "
-                                  PCMK_XE_EXPRESSION " %s because it is unused "
-                                  "when " PCMK_XA_BOOLEAN_OP " is %s", id, op);
+            if (value != NULL) { // Not possible with schema validation enabled
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/rules.c:1095]: whoops, yep that comes later :)

File: lib/common/rules.c [hunk 4]
```diff
@@ -1082,29 +1080,35 @@ pcmk__evaluate_attr_expression(const xmlNode *expression,
...
     if (source == pcmk__source_unknown) {
         // Not possible with schema validation enabled
-        // @COMPAT Fail expression once we can break backward compatibility
-        pcmk__config_warn("Expression %s has invalid " PCMK_XA_VALUE_SOURCE
-                          " value '%s', using default "
-                          "('" PCMK_VALUE_LITERAL "')", id, source_s);
-        source = pcmk__source_literal;
+        pcmk__config_err("Treating " PCMK_XE_EXPRESSION " %s as not passing "
+                         "because '%s' is not a valid " PCMK_XA_VALUE_SOURCE,
+                         id, value);
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/rules.c:1085]: thx

File: lib/common/rules.c [hunk 5]
```diff
@@ -1246,27 +1246,25 @@ pcmk__evaluate_op_expression(const xmlNode *op_expression,
...
     }
 
     // Validate operation interval
     interval_s = crm_element_value(op_expression, PCMK_META_INTERVAL);
     if (pcmk_parse_interval_spec(interval_s, &interval_ms) != pcmk_rc_ok) {
-        pcmk__config_warn("Treating " PCMK_XE_OP_EXPRESSION " %s as not "
-                          "passing because '%s' is not a valid interval",
-                          id, interval_s);
+        pcmk__config_err("Treating " PCMK_XE_OP_EXPRESSION " %s as not "
+                         "passing because '%s' is not a valid interval",
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/rules.c:1266]: might as well

**Comment by kgaillot** [comment]
updated commits and added a few on top per review

**Review by nrwahl2 — APPROVED** [review]

**Pull request merged**